frame/stake-tracker/src/lib.rs

```diff
@@ -144,7 +144,11 @@ impl<T: Config> OnStakingUpdate<T::AccountId, BalanceOf<T>> for Pallet<T> {
 			.defensive_proof("Unable to remove a nominator, perhaps it does not exist?");
 	}
 
-	fn on_unstake(_who: &T::AccountId) {}
+	fn on_unstake(_who: &T::AccountId) {
+		if T::VoterList::contains(who) {
+			defensive!("The staker should have already been removed!")
```

Comment:
@kianenigma I'd like to be able to test those defensive paths too. This can currently be achieved only through `should_panic` testing. We could leave this be for now, but perhaps it would be nice to introduce some sort of testing annotation for `defensive_*` ?
The issue now is that all of the defensive checks call `debug_assert!` which results in a panic when running in testing environment.

Reply:
will comment on the issue. For now, to the best of my knowledge, `should_panic` is as good as it gets :(